### PR TITLE
U4-9363 - Adds in localisation to new template editor

### PR DESCRIPTION
**Pull request by warrenbuckley**
Creating a PR for Per's localisation changes to the template editor just so we continue our internal development processes.

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/templates/edit.controller.js [hunk 1]
```diff
@@ -356,7 +357,7 @@
 
             vm.sectionsOverlay = {
                 view: "templatesections",
-                hasMaster: vm.template.masterTemplateAlias,
+                isMaster: vm.template.isMasterTemplate,
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/templates/edit.controller.js]: Sneaky code change - will check & test this as well :)

File: src/Umbraco.Web.UI.Client/src/views/templates/edit.controller.js [hunk 2]
```diff
@@ -457,7 +458,7 @@
                 return templateName;
 
             } else {
-                return "No master";
+                return "no master";
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/templates/edit.controller.js]: @perploug I know this is for catching an error & unlikely to be seen from what I can tell in the code - but as you have gone to hard work of changing all of these hardcoded strings. Worth doing one more?

**Comment by warrenbuckley** [comment]
Currently missing the key `[defaultdialogs_nomacros]` when opening the dialog for picking a macro when you have no macros to pick.

**Comment by warrenbuckley** [comment]
Please can you update the couple of places that need UI translation items highlighted in these screenshots @perploug 

If you can't do it just yet or busy, can you give me some of the Danish translations required so I can do this & merge this into the dev-7.6 branch for you.

master-template-picker
no-macros
query-builder

**Comment by madsrasmussen** [comment]
@warrenbuckley I have translated master template and the "no macros". I don’t think it will be possible to translate the query builder at the moment. All the values comes hardcoded from the server and it will not be pretty.

**Comment by perploug** [comment]
We can probably translated the hardcoded standard fields like createDate, they should already be in the translation files. but of course, document type names etc won’t.
I’ll have a look at how the query builder does this.
And also “query builder” is currently the danish translation of “query builder” as we do not have any name that makes sense in danish for this
Per Ploug Krogslund Umbraco HQ
On Tue, Jan 17, 2017 at 2:56 PM, Mads Rasmussen <notifications@github.com> wrote:
@warrenbuckley [https://github.com/warrenbuckley] I have translated master template and the "no macros". I don’t think it will be possible to translate the query builder at the moment. All the values comes hardcoded from the server and it will not be pretty.

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 3]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
-    <key alias="edittemplate">Rediger skabelon</key>
+    <key alias="edittemplate">Rediger skabelong</key>
+
+    <key alias="insertSections">Sektioner</key>
     <key alias="insertContentArea">Indsæt indholdsområde</key>
-    <key alias="insertContentAreaPlaceHolder">Indsæt indholdsområdemarkering</key>
-    <key alias="insertDictionaryItem">Indsæt ordbogselement</key>
-    <key alias="insertMacro">Indsæt makro</key>
-    <key alias="insertPageField">Indsæt Umbraco sidefelt</key>
+    <key alias="insertContentAreaPlaceHolder">Indsæt indholdsområde placeholder</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: This should be "Indsæt pladsholder for indholdsområde" or alternatively "Indsæt indholdsområdepladsholder", but definitively not in two words..

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 4]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+
+    <key alias="insert">Indsæt</key>
+    <key alias="insertDesc">Hvad vil du indsætte ?</key>
+
+    <key alias="insertDictionaryItem">Oversættelse</key>
+    <key alias="insertDictionaryItemDesc">Indsætter en oversætbar tekst som skifter efter det sprog som websitet vises i.</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: With commas fixed: "Indsætter en oversætbar tekst, som skifter efter det sprog, som websitet vises i."

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 5]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+
+    <key alias="insertMacro">Makro</key>
+    <key alias="insertMacroDesc">
+      En makro er et element som kan have forskellige indstillinger når det indsættes.
+      Brug det som en genbrugelig del af dit design såsom gallerier, formularer
+      og lister.
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: With commas fixed: "En makro er et element, som kan have forskellige indstillinger, når det indsættes. Brug det som en genbrugelig del af dit design såsom gallerier, formularer og lister."

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 6]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+    </key>
+
+    <key alias="insertPageField">Side værdi</key>
+    <key alias="insertPageFieldDesc">
+      Viser værdien af et felt fra den nuværende side. Kan indstilles til at bruge rekursive værdier eller
+      vise en standard værdi i tilfælde af at feltet er tomt.
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Viser værdien af et felt fra den nuværende side. Kan indstilles til at bruge rekursive værdier eller vise en **standardværdi** i tilfælde af, at feltet er tomt.

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 7]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+    </key>
+
+    <key alias="insertPartialView">Partial view</key>
+    <key alias="insertPartialViewDesc">
+      Et Partial View er et skabelon element som kan indsættes i andre skabeloner og derved
+      genbruges og deles på tværs af side-skabelonerne.
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Et Partial View er et **skabelonelement**, som kan indsættes i andre skabeloner og derved genbruges og deles på tværs af **sideskabelonerne**. 
Skabelonelement and sideskabelonerne in one word, added a comma

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 8]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+    </key>
+
     <key alias="mastertemplate">Master skabelon</key>
-    <key alias="quickGuide">Lynguide til Umbracos skabelontags</key>
+    <key alias="noMastertemplate">Ingen master skabelon</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Ingen **masterskabelon**
Masterskabelon in one word

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 9]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+      <code>@RenderSection(name)</code> element. Den underliggende skabelon skal have 
+      defineret en sektion via et <code>@section [name]{ ... }</code> element.
+      ]]>
+    </key>
+
+    <key alias="sectionName">Sektion navn</key>
+    <key alias="sectionMandatory">Sektionen er oblikatorisk</key>
+    
+    <key alias="sectionMandatoryDesc">
+      Hvis oblikatorisk, skal under-skabelonen indeholde en <code>@section</code> definition.
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Hvis **obligatorisk**, skal **underskabelonen** indeholde en <code>@section</code>-definition. 
Underskabelonen in one word, fixed spelling mistake and added a hyphen before definition

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 10]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+    <key alias="insertDictionaryItemDesc">Indsætter en oversætbar tekst som skifter efter det sprog som websitet vises i.</key>
+
+    <key alias="insertMacro">Makro</key>
+    <key alias="insertMacroDesc">
+      En makro er et element som kan have forskellige indstillinger når det indsættes.
+      Brug det som en genbrugelig del af dit design såsom gallerier, formularer
+      og lister.
+    </key>
+
+    <key alias="insertPageField">Side værdi</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: **Sideværdi**
One word

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 11]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
     <key alias="preview">Vis prøve</key>
     <key alias="styles">Styles</key>
   </area>
+
   <area alias="template">
-    <key alias="edittemplate">Rediger skabelon</key>
+    <key alias="edittemplate">Rediger skabelong</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Rediger **skabelon**
Fixed spelling mistake

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 12]
```diff
@@ -1004,17 +1005,91 @@ Mange hilsner fra Umbraco robotten
...
+    <key alias="renderSection">Indsæt en sektion</key>
+    <key alias="renderSectionDesc">
+      <![CDATA[
+      Henter indholdet af en sektion fra den underliggende skabelon ind, ved at indsætte et 
+      <code>@RenderSection(name)</code> element. Den underliggende skabelon skal have 
+      defineret en sektion via et <code>@section [name]{ ... }</code> element.
+      ]]>
+    </key>
+
+    <key alias="sectionName">Sektion navn</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: **Sektionsnavn**
One word

File: src/Umbraco.Web.UI/umbraco/config/lang/da.xml [hunk 13]
```diff
@@ -291,6 +291,7 @@
     <key alias="selectMember">Vælg medlem</key>
     <key alias="selectMemberGroup">Vælg medlemsgruppe</key>
     <key alias="noMacroParams">Der er ingen parametre for denne makro</key>
+    <key alias="noMacros">Der er ikke tilføjet nogle makroer</key>
```
Comment (arknu) [inline src/Umbraco.Web.UI/umbraco/config/lang/da.xml]: Der er ikke tilføjet **nogen** makroer
Nogen instead of nogle

**Comment by arknu** [comment]
I just added a bunch of comments to the Danish translation, fixing various small things. I hope that this is the correct way of doing it - otherwise please let me know :)

**Comment by madsrasmussen** [comment]
@arknu Thanks! I have updated the danish text with your corrections. This is a totally fine way to do it or you can send a PR targeting the branch you want to fix. Anyway you prefer - we appreciate it all 😄

**Comment by nul800sebastiaan** [comment]
Added a commit to remove the ` DA` suffix from untranslated words. If @arknu has more time to translate those that would be awesome!

**Pull request merged**